src/snapred/backend/data/LocalDataService.py

```diff
@@ -816,6 +828,15 @@ def _readGroupingMap(self, stateId: str) -> GroupingMap:
             raise FileNotFoundError(f'required grouping-schema map for state "{stateId}" at "{path}" does not exist')
         return parse_file_as(GroupingMap, path)
 
+    def readGroupingMap(self, runNumber: str):
+        # if the state exists then lookup its grouping map
+        if self.checkCalibrationFileExists(runNumber):
```

Comment:
There probably shouldnt be a case in which we interact with a state that lacks a grouping map.  It was previously not hooked into initialization but @ekapadi had added it.

Reply:
This check was added today to facilitate it working with PR #254.  It also handles the case of a user entering an errant run number.  In that case it loads the default grouping map.  If a user enters incomplete info, or looks for a state that isn't initialized, this allows them to select a default group and press "Continue", where they are directed to initialize state.

Reply:
How is the default grouping map used in this case?

Reply:
We want the user to be able to enter a run number for a state that is not initialized and click continue, so Darsh's state initialization view comes up.  In order for them to click continue, they have to select a grouping, so the dropdown has to be populated.

When the user first opens the Test Panel window, so far nothing has been initialized in the backend.
![image](https://github.com/neutrons/SNAPRed/assets/52183986/acf085fe-1448-4179-927a-6006c0a0fc76)
The grouping map is tied to the state, so the only map we can put in at first is the default one.

The user then has to enter a run number. 
![image](https://github.com/neutrons/SNAPRed/assets/52183986/e76135dc-03c7-4b2d-97fe-481f0902da27)

As soon as the user clicks away form the run number, but before clicking continue, the grouping map needs to change to reflect the available groupings in this state.

It does this by looking up the state ID for the run number, then looking up the grouping map for the state ID, and putting the appropriate list of groupings in the dropdown.

So far nothing has been initialized in the backend.

If the user at this point enters either an invalid run number, or a run number for which there is no state, then the only list of groupings we can use is the default one.

If the user enters a run number with no state, then selects a default grouping, then presses continue, only then will the user be guided to initialize the state.

The simplest thing here is to make this function to return a specific grouping map if it exists, and to otherwise return the default grouping map.

Reply:
Ah I see, populate defaults.